Autosave disabled when Ocarina of Time obtained but not Song of Time

JordanLongstaff opened the pull request:
Fixes #3160

Also moved Autosave hook file into the QoL subfolder.

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/4508594961.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/4508599846.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/4508646553.zip)
<!--- section:artifacts:end -->

JordanLongstaff committed:
Autosave disabled when Ocarina of Time obtained but not Song of Time

serprex commented on soh/soh/Enhancements/QoL/Autosave.cpp:
```diff
@@ -15,27 +16,28 @@ static uint64_t lastSaveTimestamp = GetUnixTimestamp();
 
     // Don't save when in title screen or debug file
     // Don't save the first 60 frames to not save the magic meter when it's still in the animation of filling it.
     // Don't save in Chamber of Sages and the Cutscene map because of remember save location and cutscene item gives.
```
should add comment listing this case

JordanLongstaff replied:
My bad. It's been added now.

JordanLongstaff committed:
Add explanation comment

Added a condition to prevent autosaving between obtaining the Ocarina of Time and the Song of Time.
